plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java

```diff
@@ -981,6 +981,54 @@ public void testIdentityColumns()
                         entry("delta.identity.allowExplicitInsert", false));
     }
 
+    @Test
+    public void testWritesToTableWithIdentityColumnFails()
+            throws Exception
+    {
+        String tableName = "test_identity_columns_" + randomNameSuffix();
+        Path tableLocation = catalogDir.resolve(tableName);
+        copyDirectoryContents(new File(Resources.getResource("databricks122/identity_columns").toURI()).toPath(), tableLocation);
+        assertUpdate("CALL system.register_table(CURRENT_SCHEMA, '%s', '%s')".formatted(tableName, tableLocation.toUri()));
+
+        // Disallowing all statements just in case though some statements may not unrelated to identity columns
+        assertQueryFails(
+                "INSERT INTO " + tableName + " VALUES (4, 4)",
+                "Writing to tables with identity columns is not supported");
+        assertQueryFails(
+                "UPDATE " + tableName + " SET a = 3",
+                "Writing to tables with identity columns is not supported");
+        assertQueryFails(
+                "DELETE FROM " + tableName,
+                "Writing to tables with identity columns is not supported");
+        assertQueryFails(
+                "MERGE INTO " + tableName + " t USING " + tableName + " s ON (t.a = s.a) WHEN MATCHED THEN UPDATE SET a = 1",
+                "Writing to tables with identity columns is not supported");
+    }
+
+    @Test
+    public void testIdentityColumnTableFeature()
+            throws Exception
+    {
+        String tableName = "test_identity_columns_table_feature_" + randomNameSuffix();
+        Path tableLocation = catalogDir.resolve(tableName);
+        copyDirectoryContents(new File(Resources.getResource("databricks133/identity_columns_table_feature").toURI()).toPath(), tableLocation);
+        assertUpdate("CALL system.register_table(CURRENT_SCHEMA, '%s', '%s')".formatted(tableName, tableLocation.toUri()));
+
+        // Disallowing all statements just in case though some statements may not unrelated to identity columns
```

Comment:
Is reading from table omitted intentionally (present in product test)?

Reply:
`testIdentityColumns` asserts read behavior, I think that was the reasoning